### PR TITLE
fix(1.x): re-introduce Proxy assertion methods

### PR DESCRIPTION
**Pull request by nikophil**

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -1444,12 +1444,16 @@ This library provides a *Repository Proxy* that wraps your object repositories t
...
-Your object factory's have helpful PHPUnit assertions:
+Both object proxies and your ModelFactory have helpful PHPUnit assertions:
 
 .. code-block:: php
 
     use App\Factory\PostFactory;
 
+    $post = PostFactory::createOne();
+    $post->assertPersisted();
+    $post->assertNotPersisted();
```
Comment (kbond) [inline docs/index.rst]:
```suggestion
    $post->_assertPersisted();
    $post->_assertNotPersisted();
```

We need to show the new method names, correct?

File: UPGRADE-2.0.md [hunk 2]
```diff
@@ -1,314 +1,3 @@
...
-```
-
-### Misc.
-- type-hinting to `Zenstruck\Foundry\RepositoryProxy` should be replaced by `Zenstruck\Foundry\Persistence\RepositoryDecorator`
-- type-hinting to `Zenstruck\Foundry\RepositoryAssertions` should be replaced by `Zenstruck\Foundry\Persistence\RepositoryAssertions`
-- Methods in `Zenstruck\Foundry\RepositoryProxy` do not return `Proxy` anymore, but they return the actual object
-
-
-
+Please, check the [upgrade guide on 2.x branch](https://github.com/zenstruck/foundry/blob/2.x/UPGRADE-2.0.md)
```
Comment (kbond) [inline UPGRADE-2.0.md]:
```suggestion
Please, check the [upgrade guide on the 2.x branch](https://github.com/zenstruck/foundry/blob/2.x/UPGRADE-2.0.md).
```

File: docs/index.rst (hunk 1 above)
Comment (nikophil) [reply docs/index.rst]: yes of course

**Review by kbond — APPROVED** [review]

File: src/Persistence/Proxy.php [hunk 3]
```diff
@@ -114,12 +114,16 @@ public function disableAutoRefresh(): static;
     public function withoutAutoRefresh(callable $callback): static;
 
     /**
-     * @deprecated without replacement
+     * @deprecated without method "_assertPersisted" instead
```
Comment (mmarton) [inline src/Persistence/Proxy.php:117]: a small wording error :) without => use

**Pull request merged**